uli_init/simulate.py

```diff
@@ -704,40 +700,41 @@ def build_molecule(molecule, length, para_weight):
     sequence : list
         List of the configuration sequence of the finished compound
     """
-    f = open("{}/{}.json".format(COMPOUND_DIR, molecule))
+    f = open(f"{COMPOUND_DIR}/{molecule}.json")
     mol_dict = json.load(f)
     f.close()
+
     monomer_sequence = random_sequence(para_weight, length)
-    molecule_string = "{}"
-
-    for idx, config in enumerate(monomer_sequence):
-        if idx == 0:  # append template, but not brackets
-            monomer_string = mol_dict["{}_template".format(config)]
-            if (molecule == "PEEK"):
-                # Change oxygen type on the terminal end of the polymer;
-                # needs its hydrogen.
-                monomer_string = "O" + monomer_string[1:]
-            molecule_string = molecule_string.format(monomer_string)
-            if len(monomer_sequence) == 1:
-                molecule_string = molecule_string.replace("{}", "")
-                continue
-
-        elif idx == length - 1:  # Don't use template for last iteration
-            brackets = polysmiles.count_brackets(
-                mol_dict["{}_deep_smiles".format(config)]
+    compound = Polymer()
+    para = mb.load(mol_dict["para_smiles"], smiles=True, backend="rdkit")
+    meta = mb.load(mol_dict["meta_smiles"], smiles=True, backend="rdkit")
+
+    if len(set(monomer_sequence)) == 2: # Copolymer
```

Comment:
won't this always be true? I guess in the case of ALL para or ALL meta.

Reply:
Well, if `para_weight` is 1 or 0, then sequence will either contain only `P` or only `M`, so in that case, we want to make sure only one monomer is being added.